### DISCUSSION TITLE
Uncaught ReferenceError: require is not defined

### DISCUSSION THREAD

**Discussion by tomikimi** [parcel-bundler/parcel#7836] [Q&A] [answered]
Hello 

Please i have an application i'm currently working,its built with pug for the template engine and nodejs, i am using the latest version of parcel to bundle my javascript files, but after running parcel watch command which bundles my code in my index.js file which is the source to bundle.js which is the main i keep getting the error 

Uncaught ReferenceError: require is not defined
Below is an image of my package.json file
![image](https://user-images.githubusercontent.com/39229082/158774291-077f660d-9ecf-4991-9e1a-c85c2bf691dd.png)

Below is an image of the final bundled file that throws the error
![image](https://user-images.githubusercontent.com/39229082/158774623-61cdb721-5a85-4ab1-96db-a30a89835391.png)

Please i would really appreciate the help

**Comment by mischnic** [comment, marked as answer]
The `"main": "..."` line means that Parcel is building your code as a library and doesn't include third-party dependencies. Remove that line.

**Reply by tomikimi** [reply]
Thank you its working now

Please if i want to build the project with parcel hope i won't encounter any issue
